examples/gadgets/simple/trace_open/main.go

```diff
@@ -0,0 +1,114 @@
+	// attaching them, reading maps, etc.
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf"
+
+	// The simple operator is a wrapper to make it easier to implement operators
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/simple"
+
+	// Datasources provide access to the data produced by gadgets and operators.
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+
+	// The json formatter is used to format the data as json.
+	igjson "github.com/inspektor-gadget/inspektor-gadget/pkg/datasource/formatters/json"
+
+	// The gadget context is the glue that connects all components together.
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+
+	// The runtime is the piece that will run our gadget.
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local"
+)
+
+func do() error {
+	// First of all, we need to a (simple) operator that we'll use to subscribe
+	// to events. Check the operator documentation to learn more about them.
+	// TODO: documentation
+	// Our operator should be the last of the chain to print information after
+	// all operators have been run.
+	const opPriority = 50000
+	myOperator := simple.New("myHandler", simple.OnInit(func(gadgetCtx operators.GadgetContext) error {
+		// Subscribe to all datasources and print their output as json the terminal
+		// Check the datasources documentation for more information
+		// TODO: link to documentation
+		for _, d := range gadgetCtx.GetDataSources() {
+			jsonFormatter, _ := igjson.New(d)
+
+			d.Subscribe(func(source datasource.DataSource, data datasource.Data) error {
+				jsonOutput := jsonFormatter.Marshal(data)
+				fmt.Printf("%s\n", jsonOutput)
+				return nil
+			}, opPriority)
+		}
+		return nil
+	}))
+
+	// Then, we create a gadget context instance. This is the glue that connects
+	// all operators together.
+	// Check the documentation for the gadget context here.
+	// TODO: link to documentation
+	gadgetCtx := gadgetcontext.New(
+		context.Background(),
+		"ghcr.io/inspektor-gadget/gadget/trace_open:latest",
+		gadgetcontext.WithDataOperators(
+			ocihandler.OciHandler, // pass singleton instance of the oci-handler
+			myOperator,
+		),
+	)
+
+	// After that, we need a runtime, that's the piece that will run our gadget.
+	// In this case, we use the local runtime to run the gadget in the local host.
+	// Check the runtime documentation here.
+	runtime := local.New()
+	if err := runtime.Init(nil); err != nil {
+		return fmt.Errorf("runtime init: %w", err)
+	}
+	defer runtime.Close()
+
+	// Before running the gadget, let's set some parameters.
+	// TODO: link to documentation with the parameters supported by this gadget
+	params := map[string]string{
+		// Filter only events from the root user
+		"operator.oci.ebpf.uid": "0",
+	}
+
+	// Finally, let's run our gadget
+	if err := runtime.RunGadget(gadgetCtx, nil, params); err != nil {
+		return fmt.Errorf("running gadget: %w", err)
+	}
```

Comment:
Don't you think it's a little bit strange that we have a context but we don't wait for it to be done? When using context, I always expect the following pattern:

```go
ctx := newContext()
...
go runGadget(cxt)
...
<-ctx.Done()
```

Reply:
In this case we're controlling the context, so why to wait for it to be done? Who will mark it as done if it's not the caller? 

Perhaps we can add a signal handler and so on, but I think it's too much for this simple example.